### PR TITLE
fix(model): field keys

### PR DESCRIPTION
**Pull request by smalexan**

### CODE REVIEW

File: apps/models.go [hunk 1]
```diff
@@ -83,12 +83,15 @@ type FormComponent struct {
...
-	Select    *Select    `yaml:"select,omitempty" json:"select,omitempty"`
+	Required        bool       `yaml:"required,omitempty" json:"required,omitempty"`
+	MinLength       int64      `yaml:"minLength,omitempty" json:"minLength,omitempty"`
+	MaxLength       int64      `yaml:"maxLength,omitempty" json:"maxLength,omitempty"`
+	Pattern         string     `yaml:"pattern,omitempty" json:"pattern,omitempty"`
+	TextField       *TextField `yaml:"text_field,omitempty" json:"textField,omitempty"`
+	Select          *Select    `yaml:"select,omitempty" json:"select,omitempty"`
+	ShowOnView      bool       `yaml:"show_on_view,omitempty" json:"showOnView,omitempty"`
+	ShowValueOnEdit bool       `yaml:"show_value_on_edit,omitempty" json:"showValueOnEdit,omitempty"`
+	Editable        bool       `yaml:"editable,omitempty" json:"editable,omitempty"`
```
Comment (smalexan) [inline apps/models.go:94]: what pattern do we want here for snake case vs camel case? matching https://github.com/ketch-com/rest/blob/01d0eb8f85bc518ddb1de17c992b19be2e470aa7/server/models.go#L3121

File: apps/models.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ type Select struct {
 	Placeholder string `yaml:"placeholder,omitempty" json:"placeholder,omitempty"`
 
 	// Default value will be the value for this field, before user interaction. Having a default value will override the placeholder text
-	DefaultValue string `yaml:"default,omitempty" json:"default,omitempty"`
+	DefaultValue string `yaml:"default_value,omitempty" json:"defaultValue,omitempty"`
```
Comment (sethyates) [inline apps/models.go:53]: Why?

File: apps/models.go (hunk 1 above)
Comment (sethyates) [reply apps/models.go:94]: camelCase

File: assets/schemas/manifest.json [hunk 3]
```diff
@@ -753,7 +753,7 @@
                 "description": "The placeholder text that will appear when this field is empty",
                 "default": ""
               },
-              "default": {
+              "defaultValue": {
```
Comment (sethyates) [inline assets/schemas/manifest.json:756]: why?

File: assets/schemas/manifest.json [hunk 4]
```diff
@@ -774,7 +774,7 @@
                 "description": "The placeholder text that will appear before an option is selected",
                 "default": ""
               },
-              "default": {
+              "defaultValue": {
```
Comment (sethyates) [inline assets/schemas/manifest.json:777]: why?

File: apps/models.go (hunk 2 above)
Comment (smalexan) [reply apps/models.go:53]: https://github.com/ketch-com/rest/blob/main/server/models.go#L2973 and https://github.com/ketch-com/rest/blob/main/server/models.go#L3007. Thoughts on best key name - happy to update rest

File: apps/models.go (hunk 1 above)
Comment (smalexan) [reply apps/models.go:94]: Should we go camel case across yaml and json? Inconsistent throughout - now would be the time to do it

Comment (sethyates) [reply apps/models.go:94]: yes

File: apps/models.go (hunk 2 above)
Comment (sethyates) [reply apps/models.go:53]: Why can't we keep it as `default` in the YAML file at least? Or is this not related to that?

Comment (smalexan) [reply apps/models.go:53]: We can

**Pull request closed without merging**